hamilton/function_modifiers/macros.py

```diff
@@ -324,25 +323,33 @@ class Applicable:
 
     def __init__(
         self,
-        fn: Callable,
+        fn: Union[Callable, str, None],
         args: Tuple[Union[Any, SingleDependency], ...],
         kwargs: Dict[str, Union[Any, SingleDependency]],
+        target_fn: Union[Callable, str, None] = None,
```

Comment:
Is there any case in which you would have both a `target_fn` and a `fn`?

Reply:
We have both of them in the current implementation.

1. we add add `target_fn` with `apply_to`
2. we add `fn` when we create `new_pipe_step`

So at this point the Applicable has both stored.

We then resolve for `target_fn` if we should just add the Applicable as a next step or create a new `pipe_output` to decorate the `target_fn`.